invenio_vocabularies/services/facets.py

```diff
@@ -100,36 +94,36 @@ def _vocab_to_label(self, vocab):
 
     def __call__(self, ids):
         """Return the mapping when evaluated."""
+        if not ids:
+            return {}
+
+        labels = {}
         try:
             if self.cache:
-                vocab_list = cached_vocabs(
-                    self.service_name,
-                    self.vocabulary,
-                    tuple(ids),
-                    self.fields,
-                    ttl_hash=get_ttl_hash(seconds=self.cache_ttl),
-                )
+                cache_ttl = round(time.time() / self.cache_ttl)
+                for id_ in ids:
+                    vocab = get_cached_vocab(
+                        self.service_id,
+                        self.vocabulary,
+                        self.fields,
+                        id_,
+                        ttl_hash=cache_ttl,
+                    )
+                    if not vocab:
+                        continue
+                    labels[vocab[self.id_field]] = self._vocab_to_label(vocab)
             else:
                 vocab_list = get_vocabs(
-                    self.service_name,
+                    self.service_id,
                     self.vocabulary,
+                    self.fields,
                     ids,
-                    list(self.fields),
                 )
+                for vocab in vocab_list:
+                    id_ = vocab[self.id_field]
+                    if id_ in ids:
+                        labels[id_] = self._vocab_to_label(vocab)
         except NoResultFound:
             raise FacetNotFoundError(self.vocabulary)
 
-        labels = {}
-        ids = set(ids)
-        seen = set()
-        for vocab in vocab_list:
-            # cannot loop over ids because vocab id is inside each item
```

Comment:
I don't understand this comment `# cannot loop over ids because vocab id is inside each item`... is it valid?